purchase_add_product_supplierinfo/purchase.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.model
+    def _check_product_supplierinfo(self):
+        lines = []
+        for line in self.order_line:
+            suppinfo = False
+            for seller in line.product_id.seller_ids:
+                if (self.partner_id == seller.name or
+                        self.partner_id.commercial_partner_id == seller.name):
+                    suppinfo = seller
+                    break
+            if not suppinfo:
+                lines.append((0, 0, {
+                    'name': line.name,
+                    'product_id': line.product_id.id,
+                    'to_variant': True
+                }))
+        return lines
+
+    @api.multi
+    def purchase_confirm(self):
+        self.ensure_one()
+        lines_for_update = self._check_product_supplierinfo()
+        if lines_for_update:
+            if self.partner_id.commercial_partner_id:
+                supplier_id = self.partner_id.commercial_partner_id
+            else:
+                supplier_id = self.partner_id
+            ctx = dict(
+                default_wizard_line_ids=lines_for_update,
+            )
```

Comment:
side note: this is not the efficient way of creating a dictionary in Python. It is better to use:

```
ctx = {'default_wizard_line_ids': lines_for_update}
```

Rationale: your version creates 2 dictionaries, one for the kwargs passed, and another one for the result, and the 1st one is then garbage collected, to it is better to use a litteral dictionary in this case.